### PR TITLE
Remove the need for @JsonSerialize when serializing authorization proxy objects with Jackson

### PR DESCRIPTION
**Pull request by kse-music**
Closes gh-15661

### CODE REVIEW

**Comment by kse-music** [comment]
@jzheaux I wonder why the spring-security-saml2-service-provider:opensaml4Test task  execution failed , and I didn’t make any changes.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks for the PR, @kse-music! Please see my inline feedback.

**Comment by jzheaux** [comment]
@kse-music there is a flaky test in that module. It has been disabled for now. Please try rebasing an push to your branch again.

File: core/src/main/java/org/springframework/security/authorization/method/AuthorizationAdvisorProxyFactory.java [hunk 1]
```diff
@@ -262,6 +263,18 @@ public Iterator<AuthorizationAdvisor> iterator() {
...
 
+	/**
+	 * Tag interface for
+	 * {@link org.springframework.security.authorization.method.AuthorizeReturnObject}
+	 * generated proxy object
+	 *
+	 * @author DingHao
+	 * @version 6.4
+	 */
+	public interface AuthorizeReturnObject {
```
[the remark this answers is not in the document]
Comment (kse-music) [reply core/src/main/java/org/springframework/security/authorization/method/AuthorizationAdvisorProxyFactory.java]: Done

File: docs/modules/ROOT/pages/servlet/authorization/method-security.adoc [hunk 2]
```diff
@@ -2258,6 +2258,39 @@ class User
 ----
 ======
 
+Or register `AuthorizeReturnObjectJackson2Module` to `ObjectMapper`:
```
[the remark this answers is not in the document]
Comment (kse-music) [reply docs/modules/ROOT/pages/servlet/authorization/method-security.adoc]: Done

File: core/src/test/java/org/springframework/security/authorization/AuthorizationAdvisorProxyFactoryTests.java [hunk 3]
```diff
@@ -336,6 +340,25 @@ public void setTargetVisitorIgnoreValueTypesThenIgnores() {
 		assertThat(factory.proxy(35)).isEqualTo(35);
 	}
 
+	@Test
```
[the remark this answers is not in the document]
Comment (kse-music) [reply core/src/test/java/org/springframework/security/authorization/AuthorizationAdvisorProxyFactoryTests.java:344]: Done

**Comment by kse-music** [comment]
@jzheaux I have completed the feedback, please help review it. Thanks

**Comment by jzheaux** [comment]
@kse-music, having researched this a bit further, I believe there is a more elegant way to achieve this and that is by configuring with `ProxyFactory#setOpaque(true)`. This will keep from exposing the advice methods on the CGLIB proxy altogether, preventing the cycle.

Would you be able to update this PR to set the `opaque` value to `true` for both `ProxyFactory` instances in `AuthorizationAdvisorProxyFactory`? Also, please add a test in `AuthorizationAdvisorProxyFactory` that confirms that JSON serialization works with a POJO and an out-of-the-box `ObjectMapper`.

Please also remove the entire `Working with Jackson` section as no Jackson configuration is necessary in this case.

**Comment by kse-music** [comment]
@jzheaux PR has been updated

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Thanks, @kse-music! This is now merged into `main`

**Pull request merged**